src/ycc_hull/controllers/notifications/helpers_notifications_controller.py

```diff
@@ -85,6 +92,93 @@ async def on_validate(self, task: HelperTaskDto, user: User) -> None:
         async with SmtpConnection() as smtp:
             await smtp.send_message(message)
 
+    async def send_reminders(
+        self,
+        upcoming_tasks: list[HelperTaskDto],
+        overdue_tasks: list[HelperTaskDto],
+    ) -> None:
+        if not emails_enabled(self._logger):
+            return
+
+        async with SmtpConnection() as smtp:
+            for task in upcoming_tasks:
+                await self._send_upcoming_task_reminder(task, smtp)
+                await asyncio.sleep(NOTIFICATION_DELAY_SECONDS)
+
+            overdue_tasks_by_contact_id: dict[int, list[HelperTaskDto]] = defaultdict(
+                list
+            )
+            for task in overdue_tasks:
+                overdue_tasks_by_contact_id[task.contact.id].append(task)
+
+            for _, tasks in overdue_tasks_by_contact_id.items():
+                if not tasks:
+                    continue
+
+                contact = tasks[0].contact
+                await self._send_overdue_tasks_reminder(contact, tasks, smtp)
+                await asyncio.sleep(NOTIFICATION_DELAY_SECONDS)
+
+    async def _send_upcoming_task_reminder(
+        self, task: HelperTaskDto, smtp: SmtpConnection
+    ) -> None:
+        warnings = _get_task_warnings(task)
+
+        message_builder = _task_notification_email_to_captain_and_helpers(task)
+
+        if warnings:
+            message_builder.to(task.contact)
+
+        message = message_builder.content(
+            f"""
+<p>Dear Sailors,</p>
+
+<p>This is just a quick reminder about your upcoming task.</p>
+
+{format_helper_task(task, warnings=warnings)}
+            """
+        ).build()
+        await smtp.send_message(message)
+
+    async def _send_overdue_tasks_reminder(
+        self,
+        contact: MemberPublicInfoDto,
+        tasks: list[HelperTaskDto],
+        smtp: SmtpConnection,
+    ) -> None:
+        if not tasks:
+            return
+
+        tasks_count = len(tasks)
+        n_overdue_tasks_str = (
+            f"{tasks_count} overdue task{'s' if tasks_count > 1 else ''}"
+        )
+
+        message = (
+            EmailMessageBuilder()
+            .to(contact)
+            .reply_to(contact)
+            .subject(n_overdue_tasks_str)
+            .content(
+                f"""
+<p>Dear {contact.first_name},</p>
+
+<p>This is a reminder that you are the contact for {n_overdue_tasks_str}.</p>
+
+{format_helper_tasks_list(tasks)}
+
+<p>You can:</p>
+
+<ul>
+    <li>Validate the tasks. During validation you will be asked to mark which members showed up and optionally you can leave a comment</li>
+    <li>If the task was not done before the deadline, maybe you want to extend it.</li>
```

Comment:
Hyperlink to extend?

Reply:
They are above:

![image](https://github.com/user-attachments/assets/42a017bb-55a5-4aaf-be4d-3d648d3e8d8f)